src/intelli/graph.h

```diff
@@ -375,6 +420,110 @@ class GT_INTELLI_EXPORT Graph : public Node
      */
     bool deleteConnection(ConnectionId connectionId);
 
+    /**
+     * @brief Moves the node (given by NodeId) from this graph to the target
+     * graph. Depending on the given `NodeIdPolicy` the node's id may be
+     * updated. Failing to move a node will not restore the previous state
+     * of the graph. Use a MementoDiff for this purpose. The underlying node
+     * object is guranteed to be kept alive.
+     * Note: All connections to this node are destroyed.
+     * @param nodeId Node to move to other graph
+     * @param policy Whether to update the node's id if necessary (if the
+     * node id should be kept, moving the node may fail)
+     * @return success
+     */
+    bool moveNode(NodeId nodeId,
```

Comment:
Why is the implementation in the header? We normaly only do this for template functions.

Reply:
I guess I can move it into the .cpp as well 👍